src/sql-parser/src/parser.rs

```diff
@@ -192,6 +203,193 @@ impl ParserError {
+
+impl StatementAction {
+    pub fn as_title_case(&self) -> &'static str {
+        match self {
+            StatementAction::Alter => "Alter",
+            StatementAction::Create => "Create",
+        }
+    }
+}
+
+impl ParserStatementError {
+    pub fn statement_type(&self) -> String {
+        self.statement
+            .map_or("unknown".to_string(), |s| s.to_string())
+    }
+
+    pub fn object_type(&self) -> Option<(StatementAction, AuditObjectType)> {
+        self.statement
+            .map(|s| match s {
+                StatementKind::AlterConnection => {
+                    Some((StatementAction::Alter, AuditObjectType::Connection))
+                }
+                StatementKind::AlterIndex => Some((StatementAction::Alter, AuditObjectType::Index)),
+                StatementKind::AlterRole => Some((StatementAction::Alter, AuditObjectType::Role)),
+                StatementKind::AlterSecret => {
+                    Some((StatementAction::Alter, AuditObjectType::Secret))
+                }
+                StatementKind::AlterSink => Some((StatementAction::Alter, AuditObjectType::Sink)),
+                StatementKind::AlterSource => {
+                    Some((StatementAction::Alter, AuditObjectType::Source))
+                }
+                StatementKind::CreateCluster => {
+                    Some((StatementAction::Create, AuditObjectType::Cluster))
+                }
+                StatementKind::CreateClusterReplica => {
+                    Some((StatementAction::Create, AuditObjectType::ClusterReplica))
+                }
+                StatementKind::CreateConnection => {
+                    Some((StatementAction::Create, AuditObjectType::Connection))
+                }
+                StatementKind::CreateDatabase => {
+                    Some((StatementAction::Create, AuditObjectType::Database))
+                }
+                StatementKind::CreateIndex => {
+                    Some((StatementAction::Create, AuditObjectType::Index))
+                }
+                StatementKind::CreateMaterializedView => {
+                    Some((StatementAction::Create, AuditObjectType::MaterializedView))
+                }
+                StatementKind::CreateRole => Some((StatementAction::Create, AuditObjectType::Role)),
+                StatementKind::CreateSchema => {
+                    Some((StatementAction::Create, AuditObjectType::Schema))
+                }
+                StatementKind::CreateSecret => {
+                    Some((StatementAction::Create, AuditObjectType::Secret))
+                }
+                StatementKind::CreateSink => Some((StatementAction::Create, AuditObjectType::Sink)),
+                StatementKind::CreateSource => {
+                    Some((StatementAction::Create, AuditObjectType::Source))
+                }
+                StatementKind::CreateSubsource => {
+                    Some((StatementAction::Create, AuditObjectType::Subsource))
+                }
```

Comment:
`CREATE SUBSOURCE` is internal only, and not something that users can actually type. I'd suggest just eliding this one from the mapping, and that way you don't need a new `ObjectType::Subsource`.

```suggestion
```